sphinx/locale/__init__.py

```diff
@@ -222,15 +234,7 @@ def setup(app):
     .. versionadded:: 1.8
     """
     def gettext(message: str, *args: Any) -> str:
-        if not is_translator_registered(catalog, namespace):
-            # not initialized yet
-            return _TranslationProxy(_lazy_translate, catalog, namespace, message)  # type: ignore[return-value]  # NOQA
-        else:
-            translator = get_translator(catalog, namespace)
-            if len(args) <= 1:
-                return translator.gettext(message)
-            else:  # support pluralization
-                return translator.ngettext(message, args[0], args[1])
```

Comment:
This seems like it's dropping some functionality; `ngettext` isn't referenced elsewhere in the codebase.  Needs more investigation..

Reply:
Ok, yep: `ngettext` isn't used elsewhere, and plural forms aren't used for any localized strings at the moment (nor do they have test coverage).

When used they do function correctly, and that should be maintained by the migration to the `_lazy_translate` method in this pull request.  Test coverage is probably a good idea, though -- and maybe this change should be unpicked into a separate PR.

Reply:
Potential cleanup for this opened in https://github.com/sphinx-doc/sphinx/pull/10979; resolving this thread.

Reply:
(to clarify for any email notifications: closing this thread related to `ngettext` cleanup -- not closing the entire pull request discussion thread)